lib/SILOptimizer/Utils/InstOptUtils.cpp

```diff
@@ -1187,27 +1187,13 @@ static bool shouldDestroyPartialApplyCapturedArg(SILValue arg,
   return true;
 }
 
-// *HEY YOU, YES YOU, PLEASE READ*. Even though a textual partial apply is
-// printed with the convention of the closed over function upon it, all
-// non-inout arguments to a partial_apply are passed at +1. This includes
-// arguments that will eventually be passed as guaranteed or in_guaranteed to
-// the closed over function. This is because the partial apply is building up a
-// boxed aggregate to send off to the closed over function. Of course when you
-// call the function, the proper conventions will be used.
-void swift::releasePartialApplyCapturedArg(SILBuilder &builder, SILLocation loc,
-                                           SILValue arg,
-                                           SILParameterInfo paramInfo,
-                                           InstModCallbacks callbacks) {
-  if (!shouldDestroyPartialApplyCapturedArg(arg, paramInfo,
-                                            builder.getFunction()))
-    return;
-
-  // Otherwise, we need to destroy the argument. If we have an address, we
-  // insert a destroy_addr and return. Any live range issues must have been
-  // dealt with by our caller.
-  if (arg->getType().isAddress()) {
-    // Then emit the destroy_addr for this arg
-    SILInstruction *newInst = builder.emitDestroyAddrAndFold(loc, arg);
+void swift::emitDestroyOperation(SILBuilder &builder, SILLocation loc,
+                                 SILValue operand, InstModCallbacks callbacks) {
+  // If we have an address, we insert a destroy_addr and return. Any live range
+  // issues must have been dealt with by our caller.
+  if (operand->getType().isAddress()) {
+    // Then emit the destroy_addr for this operand
+    SILInstruction *newInst = builder.emitDestroyAddrAndFold(loc, operand);
```

Comment:
emitDetroyAddrAndFold can return a nullptr. Shouldn't createNewInst only be called if it's not a nullptr?
This is not new with this PR, so obviously it can deal with nullptrs. But still, I think, just to be on the save side createNewInst should not be called with a nullptr.

Reply:
Thanks. Fixed it.